python/MooseDocs/extensions/tagging.py

```diff
@@ -0,0 +1,189 @@
+                if len(replace_str) == 0:
+                    replace_str += tag_dict_str
+                else:
+                    replace_str += "," + tag_dict_str
+
+        replace_str = "{data:[" + replace_str + "]}"
+
+        # Find the javascript file with error checking
+        js_page = self.translator.findPage(self['js_file'])
+        js_path = js_page.source
+
+        # Setup path to write content
+        js_dest_partpath = "js/" + os.path.basename(js_path)
+        js_dest_path = os.path.join(self.translator.destination, js_dest_partpath)
+
+        # Find and replace 'data:' dict structure within supplied javascript template file content,
+        # save to javascript file in destination.
+        with open(js_path,'r') as f:
+            content=f.readlines()
+            f.close()
+        content[-1]=re.sub('\{data:\[\{.+\}\}\]\}',str(replace_str), content[-1])
+        with open(js_dest_path, 'w') as f:
+            f.writelines(content)
+            f.close()
+
+
+class TaggingCommand(command.CommandComponent):
+    COMMAND= 'tag'
+    SUBCOMMAND= None
+
+    @staticmethod
+    def defaultSettings():
+        settings = command.CommandComponent.defaultSettings()
+        settings['name'] = (None, 'ID name for page and associated key:value category/label pairs.')
+        settings['pairs'] = (None, 'Key:value pairs representing categories and page-specific labels for each category.')
+        return settings
+
+    def createToken(self, parent, info, page, settings):
+        """
+        Process name and key:value pairs provided in the documentation file, check against
+        previously-processed names and keys, and add new data to global attributes.
+        """
+        if settings['name'] is None:
+            msg = "%s: No 'name' provided for page and associated tags; check markdown file. " \
+                  "This page will not be added to the tag database!"
+            LOG.error(msg, page.name)
+            name = ''
+        else:
+            name=settings['name']
+        if settings['pairs'] is None:
+            msg = "%s: No key:value pairs provided; check markdown file and add desired pairs."
+            LOG.error(msg, page.name)
+            keylist=''
+        else:
+            keylist=settings['pairs'].split()
+        mpath=re.sub(r'^.*?moose/', 'moose/', page.source)
+        entry_key_values=[]
+        for keys in keylist:
+            key_vals=keys.split(':')
+            entry_key_values.append([key_vals[0],key_vals[1]])
+
+        good_keys=[]
+        for pair in entry_key_values:
+            if pair[0] not in self.extension.allowed_keys and len(self.extension.allowed_keys) > 0:
+                msg = "%s: Provided 'key' not in allowed_keys (see config.yml); not adding the " \
+                       "following to the database: %s"
+                LOG.warning(msg, page.name, pair[0])
+            elif len(good_keys) != 0 and pair[0] in good_keys[0]:
```

Comment:
this only checks the first entry.

I think you want to check all pairs for their first entry
not sure what the one-liner is for that

Reply:
Oh good catch. This was working "well enough" that I didn't catch it in my local testing. I think the fix is to simply perform `pair[0] in good_keys` for that part of the check and Python is smart enough to look through all without extra fuss. Will try that first.

Reply:
Fixed in e916415